plugin/src/main/java/org/opensearch/ml/action/profile/MLProfileTransportAction.java

```diff
@@ -103,27 +112,45 @@ private MLProfileNodeResponse createMLProfileNodeResponse(MLProfileRequest mlPro
         Map<String, MLModelProfile> mlLocalModels = new HashMap<>();
         MLProfileInput mlProfileInput = mlProfileRequest.getMlProfileInput();
         Set<String> targetModelIds = mlProfileInput.getModelIds();
+
+        boolean isSuperAdmin = isSuperAdminUserWrapper(clusterService, client);
+        Set<String> hiddenModels = mlProfileRequest.getHiddenModelIds() != null
+            ? mlProfileRequest.getHiddenModelIds()
+            : Collections.emptySet();
+        log.info("List of hidden models from Profile transport action: {}", Arrays.toString(hiddenModels.toArray()));
         Arrays.stream(mlTaskManager.getAllTaskIds()).forEach(taskId -> {
             MLTask mlTask = mlTaskManager.getMLTask(taskId);
-            if (mlProfileInput.isReturnAllTasks() || (!mlProfileInput.emptyTasks() && mlProfileInput.getTaskIds().contains(taskId))) {
-                log.debug("Runtime task profile is found for model {}", mlTask.getModelId());
-                mlLocalTasks.put(taskId, mlTask);
-            }
-            if (mlProfileInput.isReturnAllTasks() || (!mlProfileInput.emptyModels() && targetModelIds.contains(mlTask.getModelId()))) {
-                log.debug("Runtime task profile is found for model {}", mlTask.getModelId());
-                mlLocalTasks.put(taskId, mlTask);
+            if (isSuperAdmin || !hiddenModels.contains(mlTask.getModelId())) {
+                if (mlProfileInput.isReturnAllTasks() || (!mlProfileInput.emptyTasks() && mlProfileInput.getTaskIds().contains(taskId))) {
+                    log.debug("Runtime task profile is found for model {}", mlTask.getModelId());
```

Comment:
Is this error message correct? Same with line 129

Reply:
I didn't change anything [here](https://github.com/opensearch-project/ml-commons/blob/main/plugin/src/main/java/org/opensearch/ml/action/profile/MLProfileTransportAction.java#L108-L115)